### DISCUSSION TITLE
How to get the lowest latency video output in a browser?

### DISCUSSION THREAD

**Discussion by FoxCouncil** [savonet/liquidsoap#2219] [Q&A]
```
video = single("somevideo.mp4")

*output*(video)
```

Such that I can connect a browser to that video output as quickly as possible. I was using HLS but at 30 second delay I wanted to push myself to learn how to get it as fast as possible.

I've run out of ideas and hope someone has tried to do this before with some success, I'm even willing to stitch frames with MSE through websocket if I have too.

**Comment by gilou** [comment]
srt comes to mind to get a great compromise between quality and latency… then if you want it in the browser… I'm not sure about a JS player that can handle it, but it might exist… You can achieve < 3s with RTMP already though…

**Reply by gilou** [reply]
https://github.com/AirenSoft/OvenMediaEngine / https://github.com/AirenSoft/OvenPlayer are easy hits, I haven't tried them though ;)

**Reply by FoxCouncil** [reply]
I stumbled upon SRT just after I wrote the question, it would need to be piped over WebSockets or WebRTC due to no raw sockets on browsers IIRC...

I think I might experiment with this approach first.

**Reply by smimram** [reply]
Please keep us posted, I guess that this would be helpful to many!

**Reply by FoxCouncil** [reply]
Come to think of it, I never asked, is there an example out there in the wild of a RTMP stream direct to browser, either using Harbor or some midtier (nginx eg)?

**Comment by smimram** [comment]
I'd say that [low-latency HLS](https://developer.apple.com/documentation/http_live_streaming/enabling_low-latency_hls) is the way to go. Now you have to bug @toots to tweak our HLS implementation in order to support this... ;)

**Reply by smimram** [reply]
Requested in #2246 in order not to forget.
